### DISCUSSION TITLE
How to add aria attributes to the button of the p-calendar component?

### DISCUSSION THREAD

**Discussion by mpolo205** [primefaces/community#940] [PrimeNG] [answered]
I am using the p-calendar component with the additional icon on the right side and that is defined as a button. 
If I run Lighthouse I get the following error message:
"Buttons do not have a name accessible to screen readers"
Now I want to enhance this button with aria attributes to improve the accessibility, but question is how to do that? 
In the DOM I can see, there is a classname p-datepicker-trigger that adresses the button. I try to add the attributes with javascript to the DOM element, but I did not see any effect.
Is there a way to add aria attributes to the button of the p-calendar component?

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for the feedback, we'll add a name attribute for the trigger button https://github.com/primefaces/primeng/issues/14570. If you want to use attributes outside the API section, you can possibly use ViewChild to access the relevant object. Otherwise, you will need to wait for the implementation of passthrough to be able to do it. Passthrough will allow you to pass all the desired attributes, classes, styles, and callback values to wherever you want.

**Reply by mpolo205** [reply]
Hi @cetincakiroglu 

thanks for your answer. At the moment I try to access the calendar with ViewChild(), which is not a problem. Then I try to access the nativeElement of the object, but there the only information I get back is that the nativeElement is undefined. I am not sure why this happens Maybe because of restrictions of the component itself or because of problems with processing? 
Thats my code base. For illustation I add two console.logs. The first one returns the calendar object, the second one returns undefined.
```
export class PersonComponent implements OnInit, AfterViewInit {
  @ViewChild('calendar', { static: true }) calendarBtn: ElementRef;
  ngAfterViewInit() {
    const dateBtn = this.calendarBtn.nativeElement.querySelector(
      '.p-button-icon-only'
    );
    console.log('AfterView: ', this.calendarBtn);
    console.log('AfterView: ', this.calendarBtn.nativeElement);

    if (dateBtn) {
      this.renderer.setAttribute(
        dateBtn,
        'aria-label',
        'Button klicken um Datum auszuwählen'
      );
    }
  }
}
```
I try fix that with the ngZone module and the setTimeout method. But both show no effect. Can you Cetin, or someone else give me a hint why there is a problem?

**Reply by cetincakiroglu** [reply]
Hi,

I've checked the calendar and I see that there is already an aria-label attribute set in the element with `p-datepicker-trigger` class (v17.3.2). Which version are you using?

**Reply by mpolo205** [reply]
I am using version 15.2.0.

I think in my first post the class name is not correct. I wrote that 'p-datepicker-trigger' adresses the button, I actually I use 'p-button-icon-only ' to do that.
If I add `{ read: ElementRef } `to ViewChild I can add an aria-label to the p-calendar component.
`@ViewChild('calendar', { static: true, read: ElementRef }) `

**Reply by cetincakiroglu** [reply]
v15 does not have inputs. Accessibility implementation introduced with +v16.5. You can manage to do it by viewChild as a workaround.
